src/NHibernate/Linq/NhRelinqQueryParser.cs

```diff
@@ -50,12 +53,36 @@ static NhRelinqQueryParser()
 		/// </summary>
 		/// <param name="expression">The expression to transform.</param>
 		/// <returns>The transformed expression.</returns>
+		// Since v5.3
+		[Obsolete("Please use PartialEvaluate method instead.")]
 		public static Expression PreTransform(Expression expression)
 		{
-			var partiallyEvaluatedExpression = NhPartialEvaluatingExpressionVisitor.EvaluateIndependentSubtrees(expression);
+			var partiallyEvaluatedExpression = NhPartialEvaluatingExpressionVisitor
+				.EvaluateIndependentSubtrees(
+					expression,
+					new NhEvaluatableExpressionFilter(),
+					new Dictionary<ConstantExpression, QueryVariable>());
+
 			return PreProcessor.Process(partiallyEvaluatedExpression);
 		}
 
+		/// <summary>
+		/// Applies the minimal transformations required before parameterization,
+		/// expression key computing and parsing.
+		/// </summary>
+		/// <param name="expression">The expression to transform.</param>
+		/// <returns><see cref="PartialEvaluationResult"/> that contains the transformed expression.</returns>
+		public static PartialEvaluationResult PartialEvaluate(Expression expression)
```

Comment:
I had to change the method name as it differs only in the result type.

Reply:
We can change it back to `PreTransform` when #1868 is merged as it adds session factory parameter.

Reply:
Renamed back to `PreTransform`.